tsl/src/compression/compression.h

```diff
@@ -153,4 +287,44 @@ extern DecompressionIterator *(*tsl_get_decompression_iterator_init(
 extern void update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid);
 extern void merge_chunk_relstats(Oid merged_relid, Oid compressed_relid);
 
+/* CompressSingleRowState methods */
+struct CompressSingleRowState;
+typedef struct CompressSingleRowState CompressSingleRowState;
+
+extern CompressSingleRowState *compress_row_init(int srcht_id, Relation in_rel, Relation out_rel);
+extern SegmentInfo *segment_info_new(Form_pg_attribute column_attr);
+extern bool segment_info_datum_is_in_group(SegmentInfo *segment_info, Datum datum, bool is_null);
+extern TupleTableSlot *compress_row_exec(CompressSingleRowState *cr, TupleTableSlot *slot);
+extern void compress_row_end(CompressSingleRowState *cr);
+extern void compress_row_destroy(CompressSingleRowState *cr);
+extern void row_decompressor_decompress_row(RowDecompressor *row_decompressor,
+											Tuplesortstate *tuplesortstate);
+extern int16 *compress_chunk_populate_keys(Oid in_table, const ColumnCompressionInfo **columns,
+										   int n_columns, int *n_keys_out,
+										   const ColumnCompressionInfo ***keys_out);
+extern void compress_chunk_populate_sort_info_for_column(Oid table,
+														 const ColumnCompressionInfo *column,
+														 AttrNumber *att_nums, Oid *sort_operator,
+														 Oid *collation, bool *nulls_first);
+extern PerCompressedColumn *create_per_compressed_column(TupleDesc in_desc, TupleDesc out_desc,
+														 Oid out_relid,
+														 Oid compressed_data_type_oid);
+extern void row_compressor_init(RowCompressor *row_compressor, TupleDesc uncompressed_tuple_desc,
+								Relation compressed_table, int num_compression_infos,
+								const ColumnCompressionInfo **column_compression_info,
+								int16 *column_offsets, int16 num_columns_in_compressed_table,
+								bool need_bistate, bool recompressing);
+extern void row_compressor_finish(RowCompressor *row_compressor);
+extern void populate_per_compressed_columns_from_data(PerCompressedColumn *per_compressed_cols,
+													  int16 num_cols, Datum *compressed_datums,
+													  bool *compressed_is_nulls);
+extern void row_compressor_append_sorted_rows(RowCompressor *row_compressor,
+											  Tuplesortstate *sorted_rel, TupleDesc sorted_desc);
+extern void segment_info_update(SegmentInfo *segment_info, Datum val, bool is_null);
+
+extern Oid get_compressed_chunk_index(Relation compressed_chunk,
+									  int16 *uncompressed_col_to_compressed_col,
+									  PerColumn *per_column, int n_input_columns);
```

Comment:
Your PR does not add a caller for this so this seems unnecessary.  The function is currently also static but it is only called locally in your PR so there is no reason to change that.